src/Type/ExponentiateHelper.php

```diff
@@ -95,7 +97,12 @@ private static function exponentiateConstantScalar(ConstantScalarType $base, Typ
 		}
 
 		if ($exponent instanceof ConstantScalarType) {
-			$result = $base->getValue() ** $exponent->getValue();
+			try {
+				// @ to avoid "Warning: A non-numeric value encountered"
+				$result = @pow($base->getValue(), $exponent->getValue()); // @phpstan-ignore-line
```

Comment:
I'd rather prevent the problem by checking the type of the values instead of silencing the error.

Reply:
That would require some crazy regexes. I dont think there is a simple way to simulate the magical numeric-string behaviour: https://3v4l.org/ECZjp#v8.3.6

This way we are sure we get exactly what PHP evalutes.

Reply:
Looks like `is_numeric` would do the job: https://3v4l.org/sYolO#v8.3.6

Reply:
Not really, those are both numeric, but results in different output type: https://3v4l.org/aDslS#v8.3.6

Reply:
There is a similar open PR you might can get inspiration from

https://github.com/phpstan/phpstan-src/pull/2796